test/widgets/wrapper/code_display_test.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:fluttershow_base/components/model/enum/code_display_themes.dart';
+import 'package:fluttershow_base/components/widgets/wrapper/code_display.dart';
+
+void main() {
+  Widget makeTestableWidget({
+    required String code,
+    CodeDisplayColorThemes? codeColorTheme,
+    BoxDecoration? boxDecoration,
+    TextStyle? commentTextStyle,
+    TextStyle? keywordTextStyle,
+  }) =>
+      MaterialApp(
+        home: Scaffold(
+          body: CodeDisplay(
+            code,
+            codeColorTheme: codeColorTheme,
+            boxDecoration: boxDecoration,
+            commentTextStyle: commentTextStyle,
+            keywordTextStyle: keywordTextStyle,
+          ),
+        ),
+      );
+
+  testWidgets('test widget displays formatted code', (tester) async {
+    const code = '''
+      void main() {
+        print('Hello, world!');
+      }
+    ''';
+
+    await tester.pumpWidget(
+      makeTestableWidget(code: code),
+    );
+
+    expect(find.textContaining('void main()'), findsOneWidget);
+    expect(find.textContaining("print('Hello, world!');"), findsOneWidget);
+  });
+
+  testWidgets('test widget applies custom styles and decorations',
+      (tester) async {
+    const code = '''
+      // This is a comment
+      void main() {
+        print('Hello, world!');
+      }
+    ''';
+
+    await tester.pumpWidget(
+      makeTestableWidget(
+        code: code,
+        codeColorTheme: CodeDisplayColorThemes.base16Light,
+        boxDecoration: BoxDecoration(
+          color: Colors.yellow,
+          borderRadius: BorderRadius.circular(8),
+        ),
+        commentTextStyle: const TextStyle(fontStyle: FontStyle.italic),
+        keywordTextStyle: const TextStyle(fontWeight: FontWeight.bold),
+      ),
+    );
+
+    final containerFinder = find.byKey(const Key('CodeDisplayContainer'));
+    expect(containerFinder, findsOneWidget);
+    final container = tester.widget<Container>(containerFinder);
+    expect(
+      container.decoration,
+      BoxDecoration(
```

Comment:
Instead of writing this twice, you could create a final variable in the beginning (before pumping the widget) and use the same here. This way, it would be easier.

Similar applies to all the duplicate code here.

Reply:
You are right, refactored a bit. Hope its like you imagined it now.